clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -784,4 +784,39 @@ public static long computeChecksum(ByteBuffer buffer, int start, int size) {
         return crc.getValue();
     }
 
+    /**
+     * Read data from the channel to the given byte buffer
+     * @param channel File channel containing the data to read from
+     * @param buffer The buffer into which bytes are to be transferred
+     * @param startPosition The file position at which the transfer is to begin; must be non-negative
+     *
+     * @throws IllegalArgumentException If invalid arguments were passed
+     * @throws IOException If some other I/O error occurs
+     */
+    public static void readFullyFrom(FileChannel channel, ByteBuffer buffer, long startPosition) throws IOException {
```

Comment:
Maybe the name should simply be `readFully`. And `buffer` should maybe be `destinationBuffer`. And `startPosition` can maybe just be `position` for consistency with `FileChannel.read`.

Finally, we need a mechanism to indicate that EOF has been reached. Maybe a `boolean` is good enough. The usual way of returning an `int` is a bit weird for the case where some bytes are read and then EOF is reached.